modules/core/src/ocl.cpp

```diff
@@ -2774,19 +2774,31 @@ struct Kernel::Impl
 
     void cleanupUMats()
     {
+        bool exceptionOccurred = false;
         for( int i = 0; i < MAX_ARRS; i++ )
+        {
             if( u[i] )
             {
                 if( CV_XADD(&u[i]->urefcount, -1) == 1 )
                 {
                     u[i]->flags |= UMatData::ASYNC_CLEANUP;
-                    u[i]->currAllocator->deallocate(u[i]);
+                    try
+                    {
+                        u[i]->currAllocator->deallocate(u[i]);
+                    }
+                    catch(...)
+                    {
+                        // limited by legacy before C++11
+                        exceptionOccurred = true;
```

Comment:
It makes sense to dump exception message through logging.
See `oclCleanupCallback` below.

Reply:
ok